vm/systemSmartContracts/esdt.go

```diff
@@ -1326,6 +1332,66 @@ func (e *esdt) checkSpecialRolesAccordingToTokenType(args [][]byte, token *ESDTD
 	return nil
 }
 
+func getFirstAddressWithGivenRole(token *ESDTDataV2, definedRole []byte) ([]byte, error) {
+	for _, esdtRole := range token.SpecialRoles {
+		for _, role := range esdtRole.Roles {
+			if bytes.Equal(role, definedRole) {
+				return esdtRole.Address, nil
+			}
+		}
+	}
+
+	return nil, vm.ErrElementNotFound
+}
+
+func (e *esdt) changeToMultiShardCreate(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !e.flagTransformToMultiShardCreate.IsSet() {
+		e.eei.AddReturnMessage("invalid method to call")
+		return vmcommon.FunctionNotFound
+	}
+	if len(args.Arguments) != 1 {
+		e.eei.AddReturnMessage("invalid number of arguments")
+		return vmcommon.UserError
+	}
+	token, returnCode := e.basicOwnershipChecks(args)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+	if !token.CanAddSpecialRoles {
+		e.eei.AddReturnMessage("cannot add special roles")
+		return vmcommon.UserError
+	}
+	if token.CanCreateMultiShard {
+		e.eei.AddReturnMessage("it is already multi shard create")
+		return vmcommon.UserError
+	}
+
+	addressWithCreateRole, err := getFirstAddressWithGivenRole(token, []byte(core.ESDTRoleNFTCreate))
+	if err != nil {
+		e.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	token.CanCreateMultiShard = true
+	isAddressLastByteZero := addressWithCreateRole[len(addressWithCreateRole)-1] == 0
```

Comment:
why do we need this?

Reply:
You do not need to put a new start nonce for the address which ends in zero. That address already made NFTs from 0 to X.